cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -409,18 +412,21 @@ def __pow__(self, exponent) -> 'CliffordGate':
             )
         if exponent == 1:
             return self
-        if exponent > 0 and int(exponent) == exponent:
-            base_tableau = self.clifford_tableau.copy()
-            for _ in range(int(exponent) - 1):
-                base_tableau = base_tableau.then(self.clifford_tableau)
-            return CliffordGate.from_clifford_tableau(base_tableau)
-        if exponent < 0 and int(exponent) == exponent:
-            base_tableau = self.clifford_tableau.copy()
-            for _ in range(int(-exponent) - 1):
-                base_tableau = base_tableau.then(self.clifford_tableau)
-            return CliffordGate.from_clifford_tableau(base_tableau.inverse())
 
-        return NotImplemented
+        base_tableau = self.clifford_tableau.copy()
+        if exponent < 0:
+            base_tableau = base_tableau.inverse()
+            exponent = int(abs(exponent))
```

Comment:
nit: no need to case again to `int` since we already checked that exponent is int